tests/unit/dojson/test_dojson_experiments.py

```diff
@@ -151,26 +151,65 @@ def test_experiment_names_and_affiliation_from_marcxml_119():
 
     result = strip_empty_values(experiments.do(create_record(snippet)))
 
-    assert result['affiliation'][0] == 'CERN'
+    assert result['affiliation'][0]['name'] == 'CERN'
+    assert result['affiliation'][0]['curated_relation'] == False
     assert result['experiment_names'][0]['title'] == 'CERN-ALPHA'
 
 
-def test_experiment_names_and_affiliation_from_marcxml_119_two_u():
+def test_experiment_names_and_affiliation_from_marcxml_119_a_u_z():
     snippet = (
         '<record>'
         '  <datafield tag="119" ind1=" " ind2=" ">'
-        '    <subfield code="a">LATTICE-UKQCD</subfield>'
-        '    <subfield code="u">Cambridge U.</subfield>'
-        '    <subfield code="u">Edinburgh U.</subfield>'
+        '    <subfield code="a">CERN-ALPHA</subfield>'
+        '    <subfield code="u">CERN</subfield>'
+        '    <subfield code="z">902725</subfield>'
         '  </datafield>'
         '</record>'
     )
 
     result = strip_empty_values(experiments.do(create_record(snippet)))
 
-    assert result['affiliation'] == ['Cambridge U.', 'Edinburgh U.']
+    assert result['affiliation'][0]['name'] == 'CERN'
+    assert result['affiliation'][0]['curated_relation'] == True
+    assert get_recid_from_ref(result['affiliation'][0]['record']) == 902725
+    assert result['experiment_names'][0]['title'] == 'CERN-ALPHA'
+
+
+def test_experiment_names_and_affiliation_from_marcxml_multiple_119():
+    snippet = (
+        '<record>'
+        '   <datafield tag="119" ind1=" " ind2=" ">'
+        '       <subfield code="a">LATTICE-UKQCD</subfield>'
+        '   </datafield>'
+        '   <datafield tag="119" ind1=" " ind2=" ">'
+        '       <subfield code="u">Cambridge U.</subfield>'
+        '   </datafield>'
+        '   <datafield tag="119" ind1=" " ind2=" ">'
+        '       <subfield code="u">Edinburgh U.</subfield>'
+        '       <subfield code="z">902787</subfield>'
+        '   </datafield>'
+        '   <datafield tag="119" ind1=" " ind2=" ">'
+        '       <subfield code="u">Glasgow U.</subfield>'
+        '       <subfield code="z">902823</subfield>'
+        '   </datafield>'
+        '</record>'
+    ) # record/1228417
+
+    result = strip_empty_values(experiments.do(create_record(snippet)))
```

Comment:
Have a look at the shiny new `clean_record` util: https://github.com/inspirehep/inspire-next/blob/ae2c03af6fe1f51226913dcfa1fca3d753a6ee92/inspirehep/dojson/utils/__init__.py#L259